src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray_1.Builder.cs

```diff
@@ -247,7 +248,26 @@ public void Insert(int index, T item)
             /// Adds an item to the <see cref="ICollection{T}"/>.
             /// </summary>
             /// <param name="item">The object to add to the <see cref="ICollection{T}"/>.</param>
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
             public void Add(T item)
+            {
+                int count = _count;
+                T[] elements = _elements;
+
+                if ((uint)count < (uint)elements.Length)
```

Comment:
What value do the `(uint)` casts add?

Reply:
With the `uint` casts the JIT is able to eliminate the bound checks in `elements[count]`. Cf. https://github.com/dotnet/coreclr/pull/9773

Reply:
Can you add a code comment to that effect. That sounds significant, but it's not at all clear, such that I'd be afraid someone would just remove the casts and never know they eliminated the optimization.